File: backend/src/main/kotlin/fr/amory/libris/domain/SourceEdition.kt
package fr.amory.libris.domain

data class SourceEdition(
    val isbn13: Isbn13,
    val title: String,
    val subtitle: String?,
    val authors: List<SourceAuthor>,
    val series: SourceSeries?,
    val collection: String?,
    val publisher: String?,
    val publicationYear: Int?,
    val language: String?,
    val pageCount: Int?,
    val summary: String?,
    val coverUrl: String?,
)

File: frontend/src/domain/SourceEdition.ts
export type AuthorRole = "WRITER" | "ARTIST" | "COLOURIST" | "TRANSLATOR";

export type Kind = "BOOK" | "BD" | "MANGA";

export interface SourceAuthor {
  name: string;
  role: AuthorRole;
}

export interface SourceSeries {
  name: string;
  volumeNumber: number | null;
}

export interface SourceEdition {
  isbn13: string;
  kind: Kind;
  title: string;
  subtitle: string | null;
  authors: SourceAuthor[];
  series: SourceSeries | null;
  collection: string | null;
  publisher: string | null;
  publicationYear: number | null;
  language: string | null;
  pageCount: number | null;
  summary: string | null;
  coverUrl: string | null;
}
